### ISSUE TITLE
[Bug]: dependency `tempy:3.1.0 => temp-dir@3.0.0` breaks builds on vite. No top level await

### ISSUE THREAD

**Issue by DanDroryAu** [storybookjs/storybook#27467]
### Describe the bug

I've just upgraded to `storybook@8.1.5` and I am now getting a top-level await error in my storybook run `storybook dev --quiet -p 6006`

```
Error: Transform failed with 1 error:
./node_modules/temp-dir/index.js:4:27: ERROR: Top-level await is currently not supported with the "cjs" output format
```

the node module `temp-dir` has a toplvl await and my config doesnt support it. Its a dependency of `tempy` which is a dependency of `@storybook/core-common`

my storybook config `main.ts`
```
import type { StorybookConfig } from '@storybook/react-vite';

const config: StorybookConfig = {
  stories: ['../src/**/*.mdx', '../src/**/*.stories.@(js|jsx|ts|tsx)'],

  addons: [
    '@storybook/addon-essentials',
    'storybook-addon-pseudo-states',
    '@storybook/addon-interactions',
    '@chromatic-com/storybook',
  ],

  framework: {
    name: '@storybook/react-vite',
    options: {
      legacyRootApi: true,
    },
  },

  core: {
    builder: 'storybook-builder-vite',
  },

  typescript: {
    reactDocgen: 'react-docgen',
  },

  staticDirs: ['../public'],

  docs: {},
  async viteFinal(config) {
    if (config.build) config.build.sourcemap = false;
    return config;
  },
};

export default config;
```

my `vite.config.ts`
```
/// <reference types="vitest" />
import path from 'path';
import { defineConfig, loadEnv, splitVendorChunkPlugin } from 'vite';
import react from '@vitejs/plugin-react';
import viteTsconfigPaths from 'vite-tsconfig-paths';
import svgr from 'vite-plugin-svgr';
import PeerDepsExternalPlugin from 'rollup-plugin-peer-deps-external';

export default ({ mode }) => {
  process.env = Object.assign(process.env, loadEnv(mode, process.cwd(), ''));

  return defineConfig({
    plugins: [PeerDepsExternalPlugin(), react(), viteTsconfigPaths(), svgr(), splitVendorChunkPlugin()],
    resolve: {
      alias: {
        '@': path.resolve(__dirname, './src'),
      },
    },
    server: {
      // this ensures that the browser opens upon server start
      open: true,
      // this sets a default port to 3000
      port: 3000,
    },
    define: {
      'process.env': process.env,
      global: 'window',
    },
    optimizeDeps: {
      esbuildOptions: {
        // Node.js global to browser globalThis
        define: {
          global: 'globalThis',
        },
      },
    },
    build: {
      rollupOptions: {
        output: {
          manualChunks: (id) => {
            if (id.includes('fusioncharts') && !id.includes('react-fusioncharts')) {
              return 'fusioncharts';
            }
          },
        },
      },
    },
    test: {
      globals: true,
      environment: 'jsdom',
      setupFiles: ['./src/tests/utils/setupTests.jsx'],
      exclude: ['**/src/tests/**', '**/node_modules/**', '**/cypress/**', '**/src/libs/**', '**/build/**'],
    },
  });
};
```

I've upgraded from storybook@8.0.8 which gave me grief with the hotreload disconnecting. 

If more info is required I'll happily supply.

### Reproduction link

non

### System

```bash
Storybook Environment Info:

  System:
    OS: macOS 14.3.1
    CPU: (8) arm64 Apple M1 Pro
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.3.0 - /opt/homebrew/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn <----- active
    npm: 9.6.7 - /opt/homebrew/bin/npm
  Browsers:
    Chrome: 125.0.6422.113
    Edge: 125.0.2535.79
    Safari: 17.3.1
  npmPackages:
    @storybook/addon-essentials: ^8.1.5 => 8.1.5
    @storybook/addon-interactions: ^8.1.5 => 8.1.5
    @storybook/addon-links: ^8.1.1 => 8.1.1
    @storybook/addon-mdx-gfm: ^8.1.5 => 8.1.5
    @storybook/node-logger: ^8.1.1 => 8.1.1
    @storybook/react: ^8.1.5 => 8.1.5
    @storybook/react-vite: ^8.1.5 => 8.1.5
    @storybook/test: ^8.1.5 => 8.1.5
    chromatic: ^6.24.1 => 6.24.1
    eslint-plugin-storybook: ^0.8.0 => 0.8.0
    storybook: ^8.1.5 => 8.1.5
    storybook-addon-pseudo-states: ^3.1.1 => 3.1.1
```

**Comment by DanDroryAu** [comment]
To anyone else encountering this. `storybook@8.1.4` does not use `tempy@3.0.1` which has the updated `temp-dir@3.0.0` with the top level await. I downgraded and got it running again.

**Comment by shilman** [comment]
@DanDroryAu Perhaps try deleting this line from your config?

```
  core: {
    builder: 'storybook-builder-vite',
  },
```

If that doesn't work, do you have a reproduction you can share? I just created a fresh project and am not having issues. For more info on reproductions, see https://storybook.new or see [repro docs](https://storybook.js.org/docs/react/contribute/how-to-reproduce). Thank you! 🙏

**Comment by DanDroryAu** [comment]
@shilman hey, I can't share this work repo but I'll see if I can dumb it down with reproduction later-on over the weekend. I have tried removing the builder before and to no avail. 

Reverting to 8.1.4 has unblocked me for now. 

I've not tried creating a fresh repo with 8.1.5 myself so It may be my machine or our repo thats causing it.
